<!DOCTYPE html>
<html>
  <!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
  
  <title>scGCO - Qian&#39;s blog</title>
  <link rel="icon" href="https://i.imgur.com/SrZt9VU.jpg" type="image/x-icon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
  
  <meta name="keywords" content=tools,principle,spatial transcriptomics>
  
  
  
    <link rel="alternate" href="/atom.xml " title="Qian&#39;s blog" type="application/atom+xml">
  

  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

  <body>
    <div class="container">
      <header class="header">
  <div class="blog-title">
    <a href="/" class="logo">Qian&#39;s blog</a>
    <div class="subtitle"></div>
  </div>
  <nav class="navbar">
    <ul class="menu">
      
        <li class="menu-item">
          <a href="/" class="menu-item-link">Home</a>
        </li>
      
        <li class="menu-item">
          <a href="/CV" class="menu-item-link">CV</a>
        </li>
      
        <li class="menu-item">
          <a href="/atom.xml" class="menu-item-link">RSS</a>
        </li>
      
    </ul>
  </nav>
</header>

<article class="post">
  <div class="post-title">
    <h1 class="article-title">scGCO</h1>
  </div>

  <!-- 文章目录 -->
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#Part1"><span class="toc-text">Part1</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Part1-1"><span class="toc-text">Part1</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Part2"><span class="toc-text">Part2</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Reference"><span class="toc-text">Reference</span></a></li></ol>

  <div class="post-content">
    <p>Recently, I read a fantastic literature about finding SVGs through HMMF and graph cuts algorithm named <em>Identification of spatially variable genes with graph cuts</em>.</p>
<h1 id="Part1"><a href="#Part1" class="headerlink" title="Part1"></a>Part1</h1><p>Firstly, we must figure out why finding SVGs is critical in spatial transcriptomics?</p>
<p>Firstly, SVGs are defined as genes whose expression distributions display significant dependence on their spatial locations.</p>
<h1 id="Part1-1"><a href="#Part1-1" class="headerlink" title="Part1"></a>Part1</h1><p>In this paper, it mentions three reasons:</p>
<p>1、SVGs could demonstrate strong conservation in their spatial patterns, such that many SV genes display similar dependencies on spatial locations, resulting in similar trends in spatial patterns of their expression values.</p>
<p>2、SVGs are often markers or essential regulators for tissue pattern formation and homeostasis.</p>
<p>3、Although these methods demonstrated usefulness in identifying SV genes, they cannot illustrate the exact boundaries of regions demonstrating spatial dependence, which are often of biological interest, and more importantly, these algorithms have computational efficiency of O(n2)orworseandareunsuitable for analyzing large spatial transcriptomics datasets.</p>
<h1 id="Part2"><a href="#Part2" class="headerlink" title="Part2"></a>Part2</h1><p>How does this paper identify SVGs？</p>
<p><img src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/ScGCO_Fig.%201.webp" alt=""></p>
<p>As the above picture illustrated, firstly, to model the spatial gene expression data with HMRF, scGCO first employs Delaunay triangulation to transform spatial locations of measured cells/spots into an undirected graph, where each node represents a data point (a single cell or a spot measuring multiple cells, depending on the technologies utilized) in the spatial transcriptomics data.</p>
<p>Secondly, scGCO utilizes Gaussian mixture modeling to separate a gene’s expression values into different bins, which represent different gene expression states. The float gene expression value associated with each node in the HMRF was then transformed into the corresponding bin number, and the resulting bin number was assigned to the node, creating an initial label assignment for the HMRF. </p>
<p>Thirdly, The initialized HMRF was then optimized by the fast graph cuts algorithm of Boykov to learn the true hidden labels of the nodes in HMRF, which represent authentic gene expression states.</p>
<p>Fourthly, the p-value for each gene was evaluated using the best segmentation under the complete spatial randomness (CSR) framework. Benjamini–Hochberg (BH) correction was utilized to identify spatially variable (SV) genes at the genome-scale.</p>
<h1 id="Reference"><a href="#Reference" class="headerlink" title="Reference"></a>Reference</h1><ul>
<li>[1] <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-022-33182-3">Identification of spatially variable genes with graph cuts</a></li>
</ul>

  </div>
  <div class="post-meta">
    <span class="post-time" style="color:#646060;font-size:13px">2022-06-10</span>
  </div>

  <div class="post-footer">
    
      <ul class="post-tag-list" itemprop="keywords"><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/principle/" rel="tag">principle</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/spatial-transcriptomics/" rel="tag">spatial transcriptomics</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/tools/" rel="tag">tools</a></li></ul>
    

    <a href="#top" class="top">Back to Top</a>
  </div>
</article>

<!-- 评论插件 -->

<footer>
  2023
  <span class="author">
    Qian Li
  </span>
</footer>


    </div>
  </body>
</html>