<!DOCTYPE html>
<html>
  <!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
  
  <title>Use of “default” parameter settings when analyzing single cell RNA sequencing data using Seurat a biologist’s perspective - Qian&#39;s blog</title>
  <link rel="icon" href="https://i.imgur.com/SrZt9VU.jpg" type="image/x-icon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
  
  <meta name="keywords" content=research,method>
  
  
  
    <link rel="alternate" href="/atom.xml " title="Qian&#39;s blog" type="application/atom+xml">
  

  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

  <body>
    <div class="container">
      <header class="header">
  <div class="blog-title">
    <a href="/" class="logo">Qian&#39;s blog</a>
    <div class="subtitle"></div>
  </div>
  <nav class="navbar">
    <ul class="menu">
      
        <li class="menu-item">
          <a href="/" class="menu-item-link">Home</a>
        </li>
      
        <li class="menu-item">
          <a href="/CV" class="menu-item-link">CV</a>
        </li>
      
        <li class="menu-item">
          <a href="/atom.xml" class="menu-item-link">RSS</a>
        </li>
      
    </ul>
  </nav>
</header>

<article class="post">
  <div class="post-title">
    <h1 class="article-title">Use of “default” parameter settings when analyzing single cell RNA sequencing data using Seurat a biologist’s perspective</h1>
  </div>

  <!-- 文章目录 -->
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#use-of-default-parameter-settings-when-analyzing-single-cell-rna-sequencing-data-using-seurat-a-biologists-perspective"><span class="toc-text">Use
of “default” parameter settings when analyzing single cell RNA
sequencing data using Seurat: a biologist’s perspective</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#abstract"><span class="toc-text">Abstract</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#review"><span class="toc-text">Review</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#reference"><span class="toc-text">Reference</span></a></li></ol></li></ol>

  <div class="post-content">
    <h1
id="use-of-default-parameter-settings-when-analyzing-single-cell-rna-sequencing-data-using-seurat-a-biologists-perspective">Use
of “default” parameter settings when analyzing single cell RNA
sequencing data using Seurat: a biologist’s perspective</h1>
<h2 id="abstract">Abstract</h2>
<p>Analysis of large datasets has become integral to biological studies
due to the advent of high throughput technologies such as next
generation sequencing. Techniques for analyzing these large datasets are
normally developed by bioinformaticists and statisticians, with input
from biologists. Frequently, the end-user does not have the training or
knowledge to make informed decisions on input parameter settings
required to implement the analyses pipelines. Instead, the end-user
relies on “default” settings present within the software packages,
consultations with in-house bioinformaticists, or on methods described
in previous publications. The aim of this study was to explore the
effects of altering default parameters on the cell clustering solutions
generated by a common pipeline implemented in the Seurat R package that
is used to cluster cells based on single cell RNA sequencing (scRNAseq)
data.</p>
<h2 id="review">Review</h2>
<p>However, its thinking way of default parameters could help us widen
our horizons. Acturally, we are not supposed to be confined to Seurat,
there has many other packages and algorithms to use. Also, the
comparison of result is too easy.</p>
<h2 id="reference">Reference</h2>
<ul>
<li>[1] <a target="_blank" rel="noopener" href="https://jtggjournal.com/article/view/3809">Use of
“default” parameter settings when analyzing single cell RNA sequencing
data using Seurat: a biologist's perspective</a></li>
</ul>

  </div>
  <div class="post-meta">
    <span class="post-time" style="color:#646060;font-size:13px">2023-07-09</span>
  </div>

  <div class="post-footer">
    
      <ul class="post-tag-list" itemprop="keywords"><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/method/" rel="tag">method</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/research/" rel="tag">research</a></li></ul>
    

    <a href="#top" class="top">Back to Top</a>
  </div>
</article>

<!-- 评论插件 -->

<footer>
  2023
  <span class="author">
    Qian Li
  </span>
</footer>


    </div>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

</body>
</html>