<!DOCTYPE html>
<html>
  <!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
  
  <title>Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution - Qian&#39;s blog</title>
  <link rel="icon" href="https://i.imgur.com/SrZt9VU.jpg" type="image/x-icon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
  
  <meta name="keywords" content=research,benchmarking,method>
  
  
  
    <link rel="alternate" href="/atom.xml " title="Qian&#39;s blog" type="application/atom+xml">
  

  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

  <body>
    <div class="container">
      <header class="header">
  <div class="blog-title">
    <a href="/" class="logo">Qian&#39;s blog</a>
    <div class="subtitle"></div>
  </div>
  <nav class="navbar">
    <ul class="menu">
      
        <li class="menu-item">
          <a href="/" class="menu-item-link">Home</a>
        </li>
      
        <li class="menu-item">
          <a href="/CV" class="menu-item-link">CV</a>
        </li>
      
        <li class="menu-item">
          <a href="/atom.xml" class="menu-item-link">RSS</a>
        </li>
      
    </ul>
  </nav>
</header>

<article class="post">
  <div class="post-title">
    <h1 class="article-title">Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution</h1>
  </div>

  <!-- 文章目录 -->
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#benchmarking-spatial-and-single-cell-transcriptomics-integration-methods-for-transcript-distribution-prediction-and-cell-type-deconvolution"><span class="toc-text">Benchmarking
spatial and single-cell transcriptomics integration methods for
transcript distribution prediction and cell type deconvolution</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#abstract"><span class="toc-text">Abstract</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#results"><span class="toc-text">Results</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#figure1"><span class="toc-text">Figure1</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#figure2"><span class="toc-text">Figure2</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#figure3"><span class="toc-text">Figure3</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#figure4"><span class="toc-text">Figure4</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#reference"><span class="toc-text">Reference</span></a></li></ol></li></ol>

  <div class="post-content">
    <h1
id="benchmarking-spatial-and-single-cell-transcriptomics-integration-methods-for-transcript-distribution-prediction-and-cell-type-deconvolution">Benchmarking
spatial and single-cell transcriptomics integration methods for
transcript distribution prediction and cell type deconvolution</h1>
<h2 id="abstract">Abstract</h2>
<p>Spatial transcriptomics approaches have substantially advanced our
capacity to detect the spatial distribution of RNA transcripts in
tissues, yet it remains challenging to characterize
whole-transcriptome-level data for single cells in space. Addressing
this need, researchers have developed integration methods to combine
spatial transcriptomic data with single-cell RNA-seq data to predict the
spatial distribution of undetected transcripts and/or perform cell type
deconvolution of spots in histological sections. However, to date, no
independent studies have comparatively analyzed these integration
methods to benchmark their performance. Here we present benchmarking of
16 integration methods using 45 paired datasets (comprising both spatial
transcriptomics and scRNA-seq data) and 32 simulated datasets. We found
that Tangram, gimVI, and SpaGE outperformed other integration methods
for predicting the spatial distribution of RNA transcripts, whereas
Cell2location, SpatialDWLS, and RCTD are the top-performing methods for
the cell type deconvolution of spots. We provide a benchmark pipeline to
help researchers select optimal integration methods to process their
datasets.</p>
<h2 id="results">Results</h2>
<h3 id="figure1">Figure1</h3>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/20230707_figure1.png" /></p>
<p>Figure1 shows benchmarking workflow and summary characteristics of
the examined paired datasets.</p>
<h3 id="figure2">Figure2</h3>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/20230707_figure2.png" /></p>
<p>Figure2 shows how to compare the predicting result of integration
method and the real result through different arguments.</p>
<p>“(1) structural similarity index (SSIM), which combines mean value,
variance, and covariance to measure the similarity between the predicted
result and the ground truth; (2) root mean square error (RMSE), the
absolute error between the predicted distribution and the ground truth;
and (3) Jensen–Shannon divergence (JS), which uses relative information
entropy to gauge the difference between two distributions.” (Li 等,
2022, p. 663)</p>
<p>“For one gene, a higher PCC/SSIM or lower RMSE/JS value indicates
better prediction accuracy. We also defined an accuracy score (AS) by
aggregating the four metrics (Methods) to simplify the evaluation of the
accuracy of each integration method (a higher AS value indicates better
performance).” (Li 等, 2022, p. 663)</p>
<h3 id="figure3">Figure3</h3>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/20230707_figure3.png" /></p>
<p>Figure3 compares the accuracy of the eight integration methods for
sparse spatial expression matrices down-sampled from the original
datasets.</p>
<p>Plot a shows the PCC scores of the spatial distribution of
transcripts predicted from the original data and down-sampled data from
dataset 4. The PCC values of the red-colored transcripts are greater
than 0.5 for both the original and the down-sampled data.</p>
<h3 id="figure4">Figure4</h3>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/20230707_figure4.png" /></p>
<p>Figure4 compares the performance of the 12 integration methods
capable of deconvoluting cell types of each histological spot.</p>
<h2 id="reference">Reference</h2>
<ul>
<li>[1] <a
target="_blank" rel="noopener" href="https://www.nature.com/articles/s41592-022-01480-9">Benchmarking
spatial and single-cell transcriptomics integration methods for
transcript distribution prediction and cell type deconvolution</a></li>
</ul>

  </div>
  <div class="post-meta">
    <span class="post-time" style="color:#646060;font-size:13px">2023-07-07</span>
  </div>

  <div class="post-footer">
    
      <ul class="post-tag-list" itemprop="keywords"><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/benchmarking/" rel="tag">benchmarking</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/method/" rel="tag">method</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/research/" rel="tag">research</a></li></ul>
    

    <a href="#top" class="top">Back to Top</a>
  </div>
</article>

<!-- 评论插件 -->

<footer>
  2023
  <span class="author">
    Qian Li
  </span>
</footer>


    </div>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

</body>
</html>