<!DOCTYPE html>
<html>
  <!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
  
  <title>What is cancer metabolism? - Qian&#39;s blog</title>
  <link rel="icon" href="https://i.imgur.com/SrZt9VU.jpg" type="image/x-icon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
  
  <meta name="keywords" content=tools,principle>
  
  
  
    <link rel="alternate" href="/atom.xml " title="Qian&#39;s blog" type="application/atom+xml">
  

  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

  <body>
    <div class="container">
      <header class="header">
  <div class="blog-title">
    <a href="/" class="logo">Qian&#39;s blog</a>
    <div class="subtitle"></div>
  </div>
  <nav class="navbar">
    <ul class="menu">
      
        <li class="menu-item">
          <a href="/" class="menu-item-link">Home</a>
        </li>
      
        <li class="menu-item">
          <a href="/CV" class="menu-item-link">CV</a>
        </li>
      
        <li class="menu-item">
          <a href="/atom.xml" class="menu-item-link">RSS</a>
        </li>
      
    </ul>
  </nav>
</header>

<article class="post">
  <div class="post-title">
    <h1 class="article-title">What is cancer metabolism?</h1>
  </div>

  <!-- 文章目录 -->
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#reference"><span class="toc-text">Reference</span></a></li></ol>

  <div class="post-content">
    <p>Gene Set Variation Analysis is a method to calculate genesets pathway
scores across different samples.</p>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/GSVA_1.jpg" /></p>
<p>Firstly, there has an input gene expression matrix. We perform
non-parametric kernel estimation of a specific gene's cumulative density
function.</p>
<p>In the case of microarray data, a Gaussian kernel is used:</p>
<p><span class="math inline">\(\widehat{F}_{h_i}\left(x_{i
j}\right)=\frac{1}{n} \sum_{k=1}^n \int_{-\infty}^{\frac{x_{i j}-x_{i
k}}{h_i}} \frac{1}{\sqrt{2 \Pi}} e^{-\frac{t^2}{2}} d t\)</span></p>
<p>In the case of RNA-seq data, a discrete Poisson kernel is
employed:</p>
<p><span class="math inline">\(\widehat{F}_r\left(x_{i
j}\right)=\frac{1}{n} \sum_{k=1}^n \sum_{y=0}^{x_{i j}}
\frac{e^{-\left(x_{i k}+r\right)}\left(x_{i k}+r\right)^y}{y
!}\)</span>,</p>
<p>The cumulative density function is estimated for every gene using all
samples from the above distributions. In simpler terms, a CDF value is
assigned to each gene in each sample.</p>
<p>With the CDF of all genes in each sample, we sort them from largest
to smallest.</p>
<p>Secondly, we normalize the rank values. <span
class="math inline">\(r_i\)</span> is the rank value of the gene <span
class="math inline">\(i\)</span> in sample <span
class="math inline">\(S\)</span>, <span class="math inline">\(p\)</span>
is the gene number.</p>
<p><span class="math inline">\(z_{i
j}=\left|\frac{p}{2}-r_i\right|\)</span></p>
<p>Thirdly, to calculate <span class="math inline">\(ES(s)\)</span> with
sort <span class="math inline">\(z_{i j}\)</span></p>
<p><span class="math inline">\(\nu_{j
k}(\ell)=\frac{\sum_{i=1}^{\ell}\left|r_{i j}\right|^\tau I\left(g_{(i)}
\in \gamma_k\right)}{\sum_{i=1}^p\left|r_{i j}\right|^\tau
I\left(g_{(i)} \in \gamma_k\right)}-\frac{\sum_{i=1}^{\ell}
I\left(g_{(i)} \notin
\gamma_k\right)}{p-\left|\gamma_k\right|}\)</span>,</p>
<p><span class="math inline">\(E S_{j k}^{\text {diff }}=\left|E S_{j
k}^{+}\right|-\left|E S_{j k}^{-}\right|=\max _{\ell=1, \ldots,
p}\left(0, v_{j k}(\ell)\right)-\min _{\ell=1, \ldots, p}\left(0, v_{j
k}(\ell)\right)\)</span>,</p>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/GSVA-2.webp" /></p>
<h1 id="reference">Reference</h1>
<ul>
<li>[1] <a
target="_blank" rel="noopener" href="https://towardsdatascience.com/decoding-gene-set-variation-analysis-8193a0cfda3">Decoding
Gene Set Variation Analysis</a></li>
<li>[2] <a
target="_blank" rel="noopener" href="https://www.jianshu.com/p/47f9f0107ffe">GSA、GSEA、ssGSEA、GSVA的算法原理及它们的联系与区别</a></li>
<li>[3] <a
target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3618321/">GSVA: gene
set variation analysis for microarray and RNA-Seq data</a></li>
</ul>

  </div>
  <div class="post-meta">
    <span class="post-time" style="color:#646060;font-size:13px">2022-05-02</span>
  </div>

  <div class="post-footer">
    
      <ul class="post-tag-list" itemprop="keywords"><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/principle/" rel="tag">principle</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/tools/" rel="tag">tools</a></li></ul>
    

    <a href="#top" class="top">Back to Top</a>
  </div>
</article>

<!-- 评论插件 -->

<footer>
  2023
  <span class="author">
    Qian Li
  </span>
</footer>


    </div>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

</body>
</html>