<!DOCTYPE html>
<html>
  <!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
  
  <title>SOMDE - Qian&#39;s blog</title>
  <link rel="icon" href="https://i.imgur.com/SrZt9VU.jpg" type="image/x-icon">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0">
  
  <meta name="keywords" content=tools,principle,spatial transcriptomics>
  
  
  
    <link rel="alternate" href="/atom.xml " title="Qian&#39;s blog" type="application/atom+xml">
  

  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

  <body>
    <div class="container">
      <header class="header">
  <div class="blog-title">
    <a href="/" class="logo">Qian&#39;s blog</a>
    <div class="subtitle"></div>
  </div>
  <nav class="navbar">
    <ul class="menu">
      
        <li class="menu-item">
          <a href="/" class="menu-item-link">Home</a>
        </li>
      
        <li class="menu-item">
          <a href="/CV" class="menu-item-link">CV</a>
        </li>
      
        <li class="menu-item">
          <a href="/atom.xml" class="menu-item-link">RSS</a>
        </li>
      
    </ul>
  </nav>
</header>

<article class="post">
  <div class="post-title">
    <h1 class="article-title">SOMDE</h1>
  </div>

  <!-- 文章目录 -->
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#part1"><span class="toc-text">Part1</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#part2"><span class="toc-text">Part2</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#reference"><span class="toc-text">Reference</span></a></li></ol>

  <div class="post-content">
    <p>Recently, I read a paper to identify spatial variable genes named
"<em>SOMDE: a scalable method for identifying spatially variable genes
with self-organizing map</em>."</p>
<h1 id="part1">Part1</h1>
<p>As its abstract shows:</p>
<blockquote>
<p>We present SOMDE, an efficient method for identifying SVgenes in
large-scale spatial expression data. SOMDE uses the self-organizing map
to cluster neighboring cells into nodes and then uses a Gaussian process
to fit the node-level spatial gene expression to identify SVgenes.
Experiments show that SOMDE is about 5–50 times faster than existing
methods with comparable results. The adjustable resolution of SOMDE
makes it the only way to give results in 5 min in large datasets of more
than 20 000 sequencing sites. SOMDE is available as a python package on
PyPI at https://pypi.org/project/somde for educational use.</p>
</blockquote>
<p>So, what is the self-organizing map?</p>
<p>A self-organizing map (SOM) is a type of <a
target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/Artificial_neural_network">artificial
neural network</a> (ANN) that is trained using unsupervised learning to
produce a low-dimensional (typically two-dimensional), discretized
representation of the input space of the training samples, called a map,
and is therefore, a method to make dimensionality reduction.
Self-organizing maps differ from other artificial neural networks as
they apply competitive learning instead of error-correction education
(such as backpropagation with gradient descent) because they use a
neighborhood function to preserve the topological properties of the
input space.</p>
<h1 id="part2">Part2</h1>
<p>And, how does it work in the paper?</p>
<p>Let us see this figure.</p>
<p><span class="math inline">\(N^2 \times k=C\)</span></p>
<p>where C is a constant representing the total number of data sites,
and k is the model parameter called the neighbor number, which denotes
the expected average number of original data sites each SOM node
represents. This parameter controls how condensed the learned SOM is. A
larger k means higher condensation.</p>
<p><img
src="https://raw.githubusercontent.com/yiranzhimo/Pictures/master/somde.jpeg.jpg" /></p>
<p>As it showed in (a), it initializes the value and dimension of SOM
Node weight <span class="math inline">\(M=(m_1,m_2,m_{N*N})\)</span> ,
like the orange spots. After initialization, SOM adjusts the weight of
each node toward the tissue spatial topology through a repetitive
training process using all spatial coordinates <span
class="math inline">\(X=(X_1,X_2,X_3,X_C)\)</span> as tranining
samples.</p>
<p>Cycle after cycle. After training, each data site maps to a unique
SOM node. Each SOM node represents the group of sites that are mapped to
it. Node weights in the learned SOM can be treated as new spatial
coordinates for the data sites. The new spatial locations <span
class="math inline">\(X=\left(\tilde{x}_1, \tilde{x}_2, \ldots,
\tilde{x}_{N \times N}\right)\)</span> compose the sparse topology of
the original data</p>
<p>In a SOM node, to define its gene expression as the linear
combination of the max value and average value of the gene expression in
the group of sites that the node represents. The reason for not using
average value as a meta-expression is due to the high sparsity in
large-scale data that could bias the result.</p>
<p><span class="math inline">\(\tilde{\boldsymbol{Y}}_{i, j}=\gamma
\cdot \max \left(y_{S 1}, y_{S 2}, \ldots\right)+(1-\gamma) \cdot
\operatorname{avg}\left(y_{S 1}, y_{S 2}, \ldots\right)\)</span></p>
<p>where <span class="math inline">\(y_{S 1}\)</span> is the expression
value of the gene at the data site <span class="math inline">\(x_{S
1}\)</span>. The combination ratio <span
class="math inline">\(\gamma\)</span> balances the local maximum and
mean gene expression. We use <span class="math inline">\(\gamma =
0.5\)</span> as the default in current experiments.</p>
<p>With this computational model, how to deduce SVGs?</p>
<p><span class="math inline">\(p\left(\tilde{\boldsymbol{y}} \mid
H_{\mathrm{G}}, \tilde{\boldsymbol{X}},
\Theta\right)=\mathcal{N}\left(\tilde{\boldsymbol{y}} \mid \mu \cdot 1,
\sigma_s^2 \cdot \boldsymbol{\Sigma}_{k(\tilde{\boldsymbol{x}},
\widetilde{\boldsymbol{x}} \mid \theta)}+\delta \cdot
\boldsymbol{I}\right)\)</span></p>
<p>where <span class="math inline">\(\tilde{\boldsymbol{y}}\)</span>
denotes one gene meta-expression in the SOM plane and <span
class="math inline">\(\tilde{\boldsymbol{X}}\)</span> denotes SOM node
locations.</p>
<p><span class="math inline">\(\delta \cdot \boldsymbol{I}\)</span>
indicates the non-spatial variance given by Gaussian distributed noise
in all observed gene meta-expressions. <span
class="math inline">\(\boldsymbol{\Sigma}_{k(\tilde{\boldsymbol{x}},
\widetilde{\boldsymbol{x}} \mid \theta)}\)</span> captures the spatial
variation by using a selected kernel function <span
class="math inline">\(k(\cdot)\)</span></p>
<p>Compared with the original Gaussian process model, model HG assumes
one gene meta-expression at all spatial locations has the same mean
value l so that the non-spatial variance cannot be regressed out. Thus
the maximum likelihood value of HG mainly depends on spatial
variance.</p>
<h1 id="reference">Reference</h1>
<ul>
<li>[1] <a
target="_blank" rel="noopener" href="https://medium.com/@abhinavr8/self-organizing-maps-ff5853a118d4">Self
Organizing Maps</a></li>
<li>[2] <a
target="_blank" rel="noopener" href="https://academic.oup.com/bioinformatics/article/37/23/4392/6308937">SOMDE:
a scalable method for identifying spatially variable genes with
self-organizing map</a></li>
</ul>

  </div>
  <div class="post-meta">
    <span class="post-time" style="color:#646060;font-size:13px">2022-04-20</span>
  </div>

  <div class="post-footer">
    
      <ul class="post-tag-list" itemprop="keywords"><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/principle/" rel="tag">principle</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/spatial-transcriptomics/" rel="tag">spatial transcriptomics</a></li><li class="post-tag-list-item"><a class="post-tag-list-link" href="/tags/tools/" rel="tag">tools</a></li></ul>
    

    <a href="#top" class="top">Back to Top</a>
  </div>
</article>

<!-- 评论插件 -->

<footer>
  2023
  <span class="author">
    Qian Li
  </span>
</footer>


    </div>
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

</body>
</html>